### PR TITLE
ticket #82 - feat(auth): implement password reset confirm endpoint (token + uid validation)

### PR DESCRIPTION
**Pull request by abushka110**
## What was done
* Added POST /api/auth/password-reset/confirm/ endpoint
* Implemented validation:
    * UID decoding and verification
    * token validation (expiry + single-use enforcement)
* Updated password securely using set_password()
* Revoked active sessions / refresh tokens after password change
* Added audit log entry for password reset action
* Enforced password complexity rules (aligned with registration)
* Response handling:
    * 200 — password changed successfully
    * 400 — invalid or expired token
    * 422 — weak password

## Summary by CodeRabbit

* **New Features**
  * Audit logging for password reset events now records action (requested/confirmed) and user agent; reset link format changed to separate uid and token segments.

* **Bug Fixes**
  * Improved validation error mapping and messaging for password reset confirmation (distinct 400/422 handling) and consistent password validation errors; ensures refresh tokens are revoked on successful reset.

* **Tests**
  * Expanded and hardened tests for request/confirmation flows, audit entries, token edge cases, and password-strength validation.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ITA-Dnipro/UA-4544/pull/181)

### CODE REVIEW

File: scalea/users/views.py [hunk 1]
```diff
@@ -122,35 +125,56 @@ class PasswordResetConfirmView(APIView):
...
-        raw_token = serializer.validated_data['token']
+        uid_b64 = serializer.validated_data['uid']
+        token = serializer.validated_data['token']
         password = serializer.validated_data['password']
 
         try:
-            uid_b64, token = raw_token.split('.', 1)
             uid = force_str(urlsafe_base64_decode(uid_b64))
             user = User.objects.filter(id=uid).first()
         except (ValueError, TypeError):
```
Comment (VovaD199) [inline scalea/users/views.py:159]: A very strange combination of two errors. It is worth writing them separately.

File: scalea/users/views.py [hunk 2]
```diff
@@ -122,35 +125,56 @@ class PasswordResetConfirmView(APIView):
 
     def post(self, request):
         serializer = PasswordResetConfirmSerializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
+        if not serializer.is_valid():
+            # Return 422 for password validation errors
```
Comment (VovaD199) [inline scalea/users/views.py]: missing password returns as 422, but this confuses weak password and required-field validation

Now any password error returns 422, including a missing field. According to the API spec, 422 is described specifically for weak password, not for malformed request. It is better to separate required/blank → 400, and validate_password() errors → 422.

**Review by VovaD199 — CHANGES_REQUESTED** [review]
overall good PR, please make some minor edits

**Comment by abushka110** [comment]
I missed that Issue #100 already implemented the full password reset flow. I started working on #82 earlier, before checking that #100 had already been merged, and I pushed the changes after Issue #92 was merged.

Despite the overlap, my changes add several improvements to the existing implementation:

- Extended audit logging to include reset actions ("requested" / "confirmed") and user agent
- Updated the reset link format to use separate "uid" and "token" segments
- Improved validation error handling with clearer separation between 400 Bad Request and 422 Unprocessable Entity
- Ensured refresh tokens are invalidated after a successful password reset
- Expanded test coverage
